pkg/storagev2/internal/uploadstore.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package internal
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+var (
+	// errTagIDAddressItemMarshalAddressIsZero is returned when trying
+	// to marshal a tagIDAddressItem with an address that is zero.
+	errTagIDAddressItemMarshalAddressIsZero = errors.New("marshal tagIDAddressItem: address is zero")
```

Comment:
So I think these errors need to be exported right?

The localstore might wrap the error differently based on these. But for that we would have to know about these I think.

Reply:
Maybe, but I wouldn't do it now but postpone it when needed.